infer/src/pulse/PulseFormula.ml

```diff
@@ -2749,6 +2797,27 @@ module Formula = struct
       {phi with atoms= Atom.Set.add atom phi.atoms; atoms_occurrences}
 
 
+    let add_termination_atom atom phi =
+      let atoms_occurrences =
+        Atom.fold_variables atom ~init:phi.atoms_occurrences ~f:(fun occurrences v' ->
+            AtomMapOccurrences.add v' ~occurs_in:atom occurrences )
+      in
```

Comment:
I don't think you want to change `atoms_occurrences` here at all: these track which variables appear in which atoms in `atoms`, not `raw_atom_conds`.